Building on Illumos Issue

Illumos doesn't have sys call.Flock. Could we please use syscall.FcntlFlock ?
```
# code.gitea.io/git/vendor/gopkg.in/src-d/go-billy.v4/osfs
../git/vendor/gopkg.in/src-d/go-billy.v4/osfs/os_posix.go:13:9: undefined: syscall.Flock
../git/vendor/gopkg.in/src-d/go-billy.v4/osfs/os_posix.go:13:41: undefined: syscall.LOCK_EX
../git/vendor/gopkg.in/src-d/go-billy.v4/osfs/os_posix.go:20:9: undefined: syscall.Flock
../git/vendor/gopkg.in/src-d/go-billy.v4/osfs/os_posix.go:20:41: undefined: syscall.LOCK_UN
```

--- a/osfs/os_posix.go
+++ b/osfs/os_posix.go
@@ -1,4 +1,4 @@
-// +build !windows
+// +build !windows,!solaris
 
 package osfs
 

--- a/osfs/os_solaris.go
+++ b/osfs/os_solaris.go
@@ -1,0 +1,19 @@
+// +build solaris
+
+package osfs
+
+import "syscall"
+
+func (f *file) Lock() error {
+	f.m.Lock()
+	defer f.m.Unlock()
+
+	return syscall.Flock(int(f.File.Fd()), syscall.LOCK_EX)
+}
+
+func (f *file) Unlock() error {
+	f.m.Lock()
+	defer f.m.Unlock()
+
+	return syscall.Flock(int(f.File.Fd()), syscall.LOCK_UN)
+}

